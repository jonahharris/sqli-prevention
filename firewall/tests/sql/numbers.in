>>> Sada testovacich SQL dotazu: cisla
# Binarni retezce
SELECT 0b1000001;
SELECT b'1000001';
SELECT B'1000001';
# Hexadecimalni retezce
SELECT x'61';
SELECT X'61';
SELECT 0x61;
SELECT 0 + 0xF;
# Cela a realna cisla
SELECT 10;
SELECT +10;
SELECT -10;
SELECT 0.1;
SELECT .1;
SELECT +0.1;
SELECT +.1;
SELECT 0.2E3;
SELECT .2e3;
SELECT .2e+2;
SELECT 1.1e-10;
SELECT -1.1e10;
# Bool
SELECT * FROM atbl where b=TRUE;
SELECT * FROM atbl where b=FALSE;
SELECT * FROM atbl where b=UNKNOWN;
SELECT * FROM atbl where b=NULL;
