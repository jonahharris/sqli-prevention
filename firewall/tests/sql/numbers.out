>>> Sada testovacich SQL dotazu: cisla
# Binarni retezce
STRING '***';SELECTNODATA 0 1;STMT;1;0;
STRING '***';SELECTNODATA 0 1;STMT;1;0;
STRING '***';SELECTNODATA 0 1;STMT;1;0;
# Hexadecimalni retezce
STRING '***';SELECTNODATA 0 1;STMT;1;0;
STRING '***';SELECTNODATA 0 1;STMT;1;0;
STRING '***';SELECTNODATA 0 1;STMT;1;0;
NUMBER 0;STRING '***';ADD;SELECTNODATA 0 1;STMT;1;0;
# Cela a realna cisla
NUMBER 10;SELECTNODATA 0 1;STMT;1;0;
NUMBER 10;SELECTNODATA 0 1;STMT;1;0;
NUMBER -10;SELECTNODATA 0 1;STMT;1;0;
FLOAT 0.1;SELECTNODATA 0 1;STMT;1;0;
FLOAT 0.1;SELECTNODATA 0 1;STMT;1;0;
FLOAT 0.1;SELECTNODATA 0 1;STMT;1;0;
FLOAT 0.1;SELECTNODATA 0 1;STMT;1;0;
FLOAT 200;SELECTNODATA 0 1;STMT;1;0;
FLOAT 200;SELECTNODATA 0 1;STMT;1;0;
FLOAT 20;SELECTNODATA 0 1;STMT;1;0;
FLOAT 1.1e-10;SELECTNODATA 0 1;STMT;1;0;
FLOAT -1.1e+10;SELECTNODATA 0 1;STMT;1;0;
# Bool
SELECTALL;TABLE atbl;NAME b;BOOL 1;CMP 4;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE atbl;NAME b;BOOL 0;CMP 4;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE atbl;NAME b;BOOL -1;CMP 4;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE atbl;NAME b;NULL;CMP 4;WHERE;SELECT 0 1 1;STMT;1;0;
